zebra-chain/src/verify.rs

```diff
@@ -0,0 +1,41 @@
+//! Context-free block and transaction verification for Zebra.
+//!
+//! Verification occurs in two stages:
+//!   - context-free verification of signatures, proofs, and scripts, handled by `zebra-chain`
+//!     (this module) and `zebra-script`
+//!   - context-dependent verification of the chain, handled by `zebra-consensus`
+//!
+//! Verification is CPU-bound, so it is provided via a `tower::Service` to support batching.
+
+use crate::block::Block;
+use crate::transaction::Transaction;
+use std::sync::Arc;
+
+/// A context-free block or transaction verification request.
+///
+/// This has a `Flush` variant, to allow manually flushing queued verification
+/// requests, even when the batching service is wrapped in other `tower` layers.
+#[derive(Debug)]
+// The associated data doesn't need separate documentation.
```

Comment:
associated data?

Reply:
The `block` and `transaction` items in the struct-like enum variants.

Reply:
ah, agreed, though I don't think the decision not to document the data members needs to be documented. Though if you want to leave this in I think that's fine, though I'd change the comment to not make it sound like you're talking about associated types on traits.

```suggestion
// The data members of the enum variants don't need separate documentation.
```